test/examples/ucp_client_server.c

```diff
@@ -363,19 +325,129 @@ static int parse_cmd(int argc, char *const argv[], char **server_addr)
     return 0;
 }
 
+/**
+ * Create a ucp worker on the given ucp context.
+ */
+static int init_worker(ucp_context_h ucp_context, ucp_worker_h *ucp_worker)
+{
+    ucp_worker_params_t worker_params;
+    ucs_status_t status;
+    int ret = 0;
+
+    memset(&worker_params, 0, sizeof(worker_params));
+
+    worker_params.field_mask  = UCP_WORKER_PARAM_FIELD_THREAD_MODE;
+    worker_params.thread_mode = UCS_THREAD_MODE_SINGLE;
+
+    status = ucp_worker_create(ucp_context, &worker_params, ucp_worker);
+    if (status != UCS_OK) {
+        fprintf(stderr, "failed to ucp_worker_create (%s)\n", ucs_status_string(status));
+        ret = -1;
+    }
+
+    return ret;
+}
+
+/**
+ * The callback on the server side which is invoked upon receiving a connection
+ * request from the client.
+ */
+static void server_conn_handle_cb(ucp_conn_request_h conn_request, void *arg)
+{
+    ucx_server_ctx_t *context = arg;
+    ucp_ep_h         ep;
+    ucp_ep_params_t  ep_params;
+    ucs_status_t     status;
+    ucp_worker_h     new_worker;
+    int ret;
+
+    /* Create a new worker */
+    ret = init_worker(context->ctx, &new_worker);
```

Comment:
i would not create a worker here. during init phase, create 2 workers: "connection_worker" and "data_worker", and use "data_worker" here to create an endpoint.
it's not a good practice to create worker for each new connection, so we also don't want to set such example.